src/model/pca.ts

```diff
@@ -0,0 +1,107 @@
+import { Tensor, RecursiveArray, matMul, mean, sub, div, sum, slice, divNoNan, sqrt } from '@tensorflow/tfjs-core';
+import { eigenSolve } from '../linalg';
+import { getCovarianceMatrix, getCorrelationMatrix } from '../stat/corcov';
+import { checkArray } from '../utils/validation';
+import { BaseEstimater } from "./base";
+import { getVariance } from '../stat';
+
+export type PCAMethodType = 'covariance' | 'correlation';
+
+const defaultPCAMethod = 'covariance';
+/**
+ * Parameters for PCA
+ */
+export interface PCAParameters {
+  /**
+   * The estimated number of components. If not specified,
+   * it equals the lesser value of nFeatures and nSamples
+   */
+  nComponents?: number;
```

Comment:
```suggestion
  numOfComponents?: number;
```
This might be more meaningful.

Reply:
in sklearn they use 'n_components' for this parameter